pages/box_buster.vue

```diff
@@ -0,0 +1,171 @@
+      show-empty
+      :empty-text="lighthouseFeedback"
+      sort-by="number_of_positives"
+      :sort-desc="true"
+      :fields="fields"
+    >
+      <template #table-caption>
+        <span>Box Summary:</span>
+        <span>Total of {{ total }} plates in box;</span>
+        <span>{{ total_with_maps }} plates with plates maps,</span>
+        <span>{{ total_without_maps }} without.</span>
+        <span>Total {{ total_positives }} positives.</span>
+      </template>
+    </b-table>
+  </b-container>
+</template>
+
+<script>
+import lighthouse from '../modules/lighthouse_service'
+import { getPlatesFromBoxBarcodes } from '@/modules/labwhere'
+
+// import lighthouse from '../modules/lighthouse_service'
+const countWithMap = (accumulator, plate) =>
+  accumulator + (plate.plate_map ? 1 : 0)
+const countWithoutMap = (accumulator, plate) =>
+  accumulator + (plate.plate_map ? 0 : 1)
+const sumPositives = (accumulator, plate) =>
+  accumulator + plate.number_of_positives
+const mapFormatter = (value) => (value ? 'Yes' : 'No')
+const countFormatter = (value, _key, item) => (item.plate_map ? value : 'N/A')
+const extractError = (response) => {
+  if (response.error) {
+    return response.error.message || response.error || 'Unidentified Error'
+  } else {
+    // In practice the user should never see this. However, if they do, it
+    // probably means something unexpected happened, so we'll make sure
+    // we don't just fail silently.
+    return defaultResponse.error
+  }
+}
+
+const defaultResponse = {
+  success: null,
+  error: 'No error: This should not be visible'
+}
+
+export default {
+  data() {
+    return {
+      barcode: '',
+      plates: [],
+      labwhereResponse: defaultResponse,
+      lighthouseResponse: defaultResponse,
+      fields: [
+        { key: 'plate_barcode', sortable: true },
+        {
+          key: 'plate_map',
+          sortable: true,
+          sortDirection: 'desc',
+          formatter: mapFormatter
+        },
+        {
+          key: 'number_of_positives',
+          sortable: true,
+          sortDirection: 'desc',
+          formatter: countFormatter
+        }
+      ]
+    }
+  },
+  computed: {
+    total() {
+      return this.plates.length
+    },
+    total_positives() {
+      return this.plates.reduce(sumPositives, 0)
+    },
+    total_with_maps() {
+      return this.plates.reduce(countWithMap, 0)
+    },
+    total_without_maps() {
+      return this.plates.reduce(countWithoutMap, 0)
+    },
+    labwhereState() {
+      return this.labwhereResponse.success
+    },
+    labwhereFeedback() {
+      const error = extractError(this.labwhereResponse)
+      return `${error}. Looking up barcode as plate.`
+    },
+    showTable() {
+      // We show the table if we've made a labwhere request, or have
+      // populates the plates via some other means.
+      return this.labwhereState !== null || this.plates.length !== 0
+    },
+    lighthouseFeedback() {
```

Comment:
If you look in the index.vue there is a way of using states which seems quite cool for showing comments etc. So for example you can set the status to busy when a call is in progress and error when an error has been returned. I am using it in the printing and plan to extract it into a module and component of it's own.